### PR TITLE
Comment AJAX

### PR DESCRIPTION
**Pull request by Casanova569**
Summary:
The trello card is considered approved if a detail (add comment) can be modified from the view using AJAX.

Trello Card:
Name: Ajax para comentarios
Link: https://trello.com/c/asX0WmRw/29-ajax-para-comentarios

### CODE REVIEW

File: src/Controllers/api/v1/MovieApiController.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using MvcMovie.Models;
+using MvcMovie.Repositories.Interfaces;
+
+namespace MvcMovie.Controllers
+{
+	[Route("api/v1/MovieApiController")]
```
Comment (truccomarcos) [inline src/Controllers/api/v1/MovieApiController.cs]: Fix identation

File: src/Controllers/api/v1/MovieApiController.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        public MovieApiController(IUnitOfWork unitOfWork) : base(unitOfWork)
+        {
+			this._unitOfWork = unitOfWork;
+        }
+
+        [HttpPost("Comment")]
+		[ValidateAntiForgeryToken]
+		public string Comment(CommentViewModel commentVM)
+		{
+			Comment comment = new Comment();
```
Comment (truccomarcos) [inline src/Controllers/api/v1/MovieApiController.cs]: Try creating a comment constructor that receives a view model

File: src/Controllers/api/v1/MovieApiController.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+			comment.Title = commentVM.Title;
+			comment.Content = commentVM.Content;
+			comment.ReleaseDate = DateTime.Now;
+			comment.MovieID = commentVM.MovieId;
+
+		    UnitOfWork.Comments.Update(comment);
+		    UnitOfWork.Complete();
+            return ("Se agrego con éxito");
```
Comment (truccomarcos) [inline src/Controllers/api/v1/MovieApiController.cs]: This should return an okay status code. Try returning `Ok(message)`

File: src/wwwroot/js/site.js [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+        var comment = new Object();
+        comment.Title=$('#comTitle').val();
+        comment.Content=$('#comContent').val();
+        comment.ReleaseDate = new Date();
+        var dataC = (comment.ReleaseDate.getMonth() + 1) + '/' + comment.ReleaseDate.getDate() + '/' +  comment.ReleaseDate.getFullYear();
+        comment.MovieId =$('[name=comMovieId]').val();
+        var tokenVal = $('[name=__RequestVerificationToken]').val();
+
+        if (comment.Title != '' && comment.Content !='')
+            tableBody.append('<tr> <td>' + comment.Title + '</td>' + '<td>' + comment.Content + '</td>' + '<td>' + dataC + '</td> <tr>');
```
Comment (truccomarcos) [inline src/wwwroot/js/site.js]: You are appending the body before you have confirmation that it was added to the database. Try appending it when the ajax is successful

**Review by truccomarcos — CHANGES_REQUESTED** [review]

File: src/Controllers/MovieAppController.cs [hunk 5]
```diff
@@ -176,35 +175,14 @@ public IActionResult Details(int id)
...
-                comment.Content = content;
-		        comment.CreatedAt = DateTime.Now;
-                comment.MovieId = id;
-		        UnitOfWork.Comments.Update(comment);
-		        UnitOfWork.Complete();
-		        return RedirectToAction("Index");
-		    }
-		    return View();
-		}
+            var comments = UnitOfWork.Comments.GetAll();
```
Comment (truccomarcos) [inline src/Controllers/MovieAppController.cs]: Why are you getting the comments here?

**Review by truccomarcos — CHANGES_REQUESTED** [review]

**Comment by truccomarcos** [comment]
Try adding a validation on the js code, so that an error is shown if the comment is empty

File: src/Controllers/MovieAppController.cs [hunk 6]
```diff
@@ -176,35 +175,14 @@ public IActionResult Details(int id)
...
+            MovieViewModel movieVM = new MovieViewModel(movie);
+            return View(movieVM);
```
Comment (mikolaslorant) [inline src/Controllers/MovieAppController.cs]: these two lines could be written as one return new View(new MovieViewModel(movie));

File: src/Controllers/api/v1/MovieApiController.cs [hunk 7]
```diff
@@ -7,26 +7,39 @@
...
         }
 
+        [HttpPost("Comment")]
+        [ValidateAntiForgeryToken]
+        public string Comment(CommentViewModel commentVM)
+        {
+            Comment comment = new Comment(commentVM);
+            UnitOfWork.Comments.Update(comment);
+            UnitOfWork.Complete();
+            return ("Ok");
```
Comment (mikolaslorant) [inline src/Controllers/api/v1/MovieApiController.cs]: Use Ok() function

File: src/Controllers/api/v1/MovieApiController.cs [hunk 8]
```diff
@@ -7,26 +7,39 @@
...
         public MovieApiController(IUnitOfWork unitOfWork, IEmailService mailer) : base(unitOfWork, mailer)
         {
         }
 
+        [HttpPost("Comment")]
+        [ValidateAntiForgeryToken]
+        public string Comment(CommentViewModel commentVM)
+        {
+            Comment comment = new Comment(commentVM);
+            UnitOfWork.Comments.Update(comment);
```
Comment (mikolaslorant) [inline src/Controllers/api/v1/MovieApiController.cs]: Why an  update and not an add?

File: src/Models/Comment.cs [hunk 9]
```diff
@@ -23,7 +23,6 @@ public class Comment
         public int MovieId { get; set; }
         public virtual Movie Movie { get; set; }
         public Comment(){ }
```
Comment (mikolaslorant) [inline src/Models/Comment.cs]: add a space after comment()

File: src/Views/MovieApp/Index.cshtml [hunk 10]
```diff
@@ -6,7 +6,7 @@
 
 <h2>Movie App</h2>
 <a asp-action="Create">Create New</a>
-<form asp-controller="Movies" asp-action="Index" method="get">
+<form asp-controller=" asp-action="Index" method="get">
```
Comment (mikolaslorant) [inline src/Views/MovieApp/Index.cshtml]: something went wrong here after the asp-controller.

File: src/src.csproj [hunk 11]
```diff
@@ -14,4 +14,4 @@
     <PackageReference Include="LocalizationCultureCore" Version="1.1.2" />
     <PackageReference Include="Swashbuckle.AspNetCore" Version="4.0.1" />
   </ItemGroup>
-</Project>
+</Project>
```
Comment (mikolaslorant) [inline src/src.csproj]: add eof

File: src/Controllers/api/v1/MovieApiController.cs [hunk 12]
```diff
@@ -7,26 +7,39 @@
...
     {
 
-		private readonly IUnitOfWork _unitOfWork;
+        private readonly IUnitOfWork _unitOfWork;
+
         private readonly IEmailService _mailer;
-        
+
         public MovieApiController(IUnitOfWork unitOfWork, IEmailService mailer) : base(unitOfWork, mailer)
         {
```
Comment (mikolaslorant) [inline src/Controllers/api/v1/MovieApiController.cs]: put these curly brackets in the upper liner

**Review by mikolaslorant — CHANGES_REQUESTED** [review]

**Review by truccomarcos — APPROVED** [review]

**Pull request merged**